src/main/java/ch/njol/skript/expressions/ExprTransform.java

```diff
@@ -114,13 +99,62 @@ public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean is
 			.iterator();
 	}
 
+	@Override
+	public Iterator<Map.Entry<String, Object>> keyedIterator(Event event) {
+		Iterator<? extends Map.Entry<String, ?>> iterator = ((KeyProviderExpression<?>) unmappedObjects).keyedIterator(event);
+		Stream<? extends Map.Entry<String, ?>> stream = StreamSupport.stream(
+			Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED),
+			false
+		);
+
+		if (!mappingExpr.isSingle()) {
+			int[] index = {0};
+			//noinspection unchecked,rawtypes
+			return (Iterator) stream
+				.flatMap(entry -> {
+					currentValue = entry.getValue();
+					currentIndex = entry.getKey();
+					return mappingExpr.stream(event);
+				})
+				.map(mapped -> Map.entry(++index[0] + "", mapped))
+				.iterator();
```

Comment:
Should we even be allowing this to have a keyed iterator if it's not a 1:1 map? I feel it's a bit misleading for the expr to say "yes i support returning keys" but actually completely clobber them.
Also, i'm sure the array index thing is for some esoteric stream reason, but it's probably worth it to add a comment elaborating.